backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java

```diff
@@ -348,6 +348,10 @@ public ExperimentRun insertExperimentRun(ExperimentRun experimentRun, UserInfo u
     checkIfEntityAlreadyExists(experimentRun, true);
     createRoleBindingsForExperimentRun(experimentRun, userInfo);
     try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      if (experimentRun.getDatasetsCount() > 0 && app.isPopulateConnectionsBasedOnPrivileges()) {
```

Comment:
the log operation can not be a silent operation and should error out if the dataset being logged is not accessible. If there are multiple datasets, we should error out even is anyone dataset is not accessible.

cc: @conradoverta if he has different though

Reply:
error out https://github.com/VertaAI/modeldb/pull/1403/commits/a1ba765876c8e268b0aaf0485b82abf115b80127